server/src/main/java/org/elasticsearch/common/util/concurrent/EsRejectedExecutionHandler.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.util.concurrent;
+
+import org.elasticsearch.common.metrics.CounterMetric;
+
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public abstract class EsRejectedExecutionHandler implements RejectedExecutionHandler {
+
+    private final CounterMetric rejected = new CounterMetric();
+
+    /**
+     * The number of rejected executions.
+     */
+    public long rejected() {
+        return rejected.count();
+    }
+
+    protected final EsRejectedExecutionException newRejectedException(Runnable r, ThreadPoolExecutor executor, boolean isExecutorShutdown) {
+        rejected.inc();
```

Comment:
For what purpose the inc() method is called here? I mean the method name is **newRejectedException**, but in the method implementation we also have an increment. I think its not so clear.

Reply:
I agree, I pushed 6e2d87acd4e4b7487d9104168b2f68d7b21d9b5a